drivers/can/can_mcan.c

```diff
@@ -297,19 +297,20 @@ int can_mcan_init(const struct device *dev, const struct can_mcan_config *cfg,
 		(can->crel & CAN_MCAN_CREL_DAY) >> CAN_MCAN_CREL_DAY_POS);
 
 #ifndef CONFIG_CAN_STM32FD
-	can->sidfc = ((uint32_t)msg_ram->std_filt & CAN_MCAN_SIDFC_FLSSA_MSK) |
+	uint32_t mrba = (uint32_t)msg_ram;
```

Comment:
Can you leave this code as it is and make your own implementation guarded with your Kconfig symbol?
We will refactor this part when all new frontends are merged.
The default for m_can is the current implementation IIRC.

Reply:
I think we need to migrate to something like `bosch,mram-cfg` in https://github.com/torvalds/linux/blob/master/Documentation/devicetree/bindings/net/can/bosch%2Cm_can.yaml going forward.

Since we don't have any M_CAN frontends using absolute addressing as of yet, I would prefer to eliminate the dead code and default to using relative addressing. We have no way to test or even compile the absolute addressing code.

Reply:
Give me a sec. PR for SAM is incoming.

Reply:
#41674

Reply:
I took a pass at this which allows for all three options.